src/components/dialog/dialog.component.js

```diff
@@ -154,20 +145,17 @@ class Dialog extends Modal {
   }
 
   get closeIcon() {
-    if (this.props.showCloseIcon) {
-      return (
-        <Icon
-          className='carbon-dialog__close'
-          data-element='close'
-          onClick={ this.props.onCancel }
-          type='close'
-          tabIndex='0'
-          role='button'
-          onKeyDown={ this.onButtonKeyDown }
-        />
-      );
-    }
-    return null;
+    const { showCloseIcon, onCancel } = this.props;
+    if (!showCloseIcon || !onCancel) return null;
+
+    return (
+      <IconButton
```

Comment:
Adding `IconButton` means that if you press `ESC` on the Icon button, but `disableEscKey=true` the modal still closes. Need to check with UX what keyboard events we should listen to, and if we should still support `disableEscKey`.

Reply:
Ben confirmed the following:

If focussed on 'X', enter or Esc closes the dialog in all situations. (Space is sometimes used for selection, but more often in radios/checkboxes, not so much for links).
If not focussed on 'X', Esc closes the dialog, unless disableEscKey is applied, in which case it does nothing.